sklearn/datasets/tests/test_samples_generator.py

```diff
@@ -184,6 +184,40 @@ def test_make_classification_informative_features():
         make(n_features=2, n_informative=2, n_classes=3, n_clusters_per_class=2)
 
 
+def test_make_classification_return_x_y():
+    """
+    Test that make_classification returns a Bunch when return_X_y is False.
+
+    Also that bunch.X is the same as X
+    """
+
+    make = partial(
+        make_classification,
+        n_samples=100,
+        n_features=20,
+        n_informative=5,
+        n_redundant=1,
+        n_repeated=1,
+        n_classes=3,
+        n_clusters_per_class=1,
+        random_state=0,
+    )
+
+    bunch = make(return_X_y=False)
+
+    assert (
+        hasattr(bunch, "DESCR")
+        and hasattr(bunch, "parameters")
+        and hasattr(bunch, "feature_info")
+        and hasattr(bunch, "X")
+        and hasattr(bunch, "y")
+    )
+
+    X, y = make()
+
+    assert np.array_equal(X, bunch.X) and np.array_equal(y, bunch.y)
```

Comment:
could you please also test for the correctness of other values included in the bunch